src/ripple/app/tx/impl/LocalTxs.cpp

```diff
@@ -61,48 +64,83 @@ class LocalTx
     static int const holdLedgers = 5;
 
     LocalTx (LedgerIndex index, STTx::ref txn)
-        : m_txn (txn)
-        , m_expire (index + holdLedgers)
-        , m_id (txn->getTransactionID ())
-        , m_account (txn->getSourceAccount ())
-        , m_seq (txn->getSequence())
+        : txn_ (txn)
+        , expire_ (index + holdLedgers)
+        , id_ (txn->getTransactionID ())
+        , accountID_ (txn->getSourceAccount().getAccountID())
+        , seq_ (txn->getSequence())
+        , ticketOwnerID_ (0)
+        , ticketSeq_ (0)
+        , ticketIndex_ ()
     {
+        if (txn->isFieldPresent (sfTicketID))
+        {
+            auto const& ticketID = txn->getFieldObject (sfTicketID);
+            ticketOwnerID_ = ticketID.getFieldAccount160 (sfAccount);
+            ticketSeq_ = ticketID.getFieldU32 (sfSequence);
+            ticketIndex_ = ripple::getTicketIndex (ticketOwnerID_, ticketSeq_);
+        }
+
         if (txn->isFieldPresent (sfLastLedgerSequence))
-            m_expire = std::min (m_expire, txn->getFieldU32 (sfLastLedgerSequence) + 1);
+            expire_ = std::min (
+                expire_, txn->getFieldU32 (sfLastLedgerSequence) + 1);
     }
 
     uint256 const& getID () const
     {
-        return m_id;
+        return id_;
     }
 
     std::uint32_t getSeq () const
     {
-        return m_seq;
+        return seq_;
     }
 
     bool isExpired (LedgerIndex i) const
     {
-        return i > m_expire;
+        return i > expire_;
     }
 
     STTx::ref getTX () const
     {
-        return m_txn;
+        return txn_;
+    }
+
+    AccountID const& getAccountID () const
```

Comment:
Just out of curiosity, and I'm not suggesting that we change this because in reality the `LocalTxs` class will go away, but what did we gain by making these accessor functions instead of just making the data members `public`?